firmware/targets/f7/furi_hal/furi_hal_nfc_iso15693.c

```diff
@@ -0,0 +1,463 @@
+    size_t fifo_buf_bits;
+    uint8_t frame_buf[FURI_HAL_NFC_ISO15693_POLLER_MAX_BUFFER_SIZE * 2];
+    size_t frame_buf_bits;
+} FuriHalNfcIso15693Poller;
+
+static FuriHalNfcIso15693Listener* furi_hal_nfc_iso15693_listener = NULL;
+static FuriHalNfcIso15693Poller* furi_hal_nfc_iso15693_poller = NULL;
+
+static FuriHalNfcIso15693Listener* furi_hal_nfc_iso15693_listener_alloc() {
+    FuriHalNfcIso15693Listener* instance = malloc(sizeof(FuriHalNfcIso15693Listener));
+
+    instance->signal = iso15693_signal_alloc(&gpio_spi_r_mosi);
+    instance->parser =
+        iso15693_parser_alloc(&gpio_spi_r_miso, FURI_HAL_NFC_ISO15693_MAX_FRAME_SIZE);
+
+    return instance;
+}
+
+static void furi_hal_nfc_iso15693_listener_free(FuriHalNfcIso15693Listener* instance) {
+    furi_assert(instance);
+
+    iso15693_signal_free(instance->signal);
+    iso15693_parser_free(instance->parser);
+
+    free(instance);
+}
+
+static FuriHalNfcIso15693Poller* furi_hal_nfc_iso15693_poller_alloc() {
+    FuriHalNfcIso15693Poller* instance = malloc(sizeof(FuriHalNfcIso15693Poller));
+
+    return instance;
+}
+
+static void furi_hal_nfc_iso15693_poller_free(FuriHalNfcIso15693Poller* instance) {
+    furi_assert(instance);
+
+    free(instance);
+}
+
+static FuriHalNfcError furi_hal_nfc_iso15693_common_init(FuriHalSpiBusHandle* handle) {
+    // Common NFC-V settings, 26.48 kbps
+
+    // 1st stage zero = 12 kHz, 3rd stage zero = 80 kHz, low-pass = 600 kHz
+    st25r3916_write_reg(
+        handle,
+        ST25R3916_REG_RX_CONF1,
+        ST25R3916_REG_RX_CONF1_z12k | ST25R3916_REG_RX_CONF1_h80 |
+            ST25R3916_REG_RX_CONF1_lp_600khz);
+
+    // Enable AGC
+    // AGC Ratio 6
+    // AGC algorithm with RESET (recommended for ISO15693)
+    // AGC operation during complete receive period
+    // Squelch automatic activation on TX end
+    st25r3916_write_reg(
+        handle,
+        ST25R3916_REG_RX_CONF2,
+        ST25R3916_REG_RX_CONF2_agc6_3 | ST25R3916_REG_RX_CONF2_agc_m |
+            ST25R3916_REG_RX_CONF2_agc_en | ST25R3916_REG_RX_CONF2_sqm_dyn);
+
+    // HF operation, full gain on AM and PM channels
+    st25r3916_write_reg(handle, ST25R3916_REG_RX_CONF3, 0x00);
+    // No gain reduction on AM and PM channels
+    st25r3916_write_reg(handle, ST25R3916_REG_RX_CONF4, 0x00);
+
+    // Collision detection level 53%
+    // AM & PM summation before digitizing on
+    st25r3916_write_reg(
+        handle,
+        ST25R3916_REG_CORR_CONF1,
+        ST25R3916_REG_CORR_CONF1_corr_s0 | ST25R3916_REG_CORR_CONF1_corr_s1 |
+            ST25R3916_REG_CORR_CONF1_corr_s4);
+    // 424 kHz subcarrier stream mode on
+    st25r3916_write_reg(handle, ST25R3916_REG_CORR_CONF2, ST25R3916_REG_CORR_CONF2_corr_s8);
+    return FuriHalNfcErrorNone;
+}
+
+static FuriHalNfcError furi_hal_nfc_iso15693_poller_init(FuriHalSpiBusHandle* handle) {
+    furi_assert(furi_hal_nfc_iso15693_poller == NULL);
+
+    furi_hal_nfc_iso15693_poller = furi_hal_nfc_iso15693_poller_alloc();
```

Comment:
```suggestion
    furi_hal_nfc_iso15693_poller_alloc();
```
A bit neat if hide assert and assign inside poller_alloc() function. This also applies to others poller_alloc and listener_alloc

Reply:
Disagree. The logic of alloc() function in our code is to allocate, initialize some structure and return the pointer. Asserts in furi_hal_nfc_iso15693_poller_init() and furi_hal_nfc_iso15693_poller_deinit() functions serve to ensure consistent calls of these functions.